src/extension/background-script/index.js

```diff
@@ -14,11 +14,21 @@ setInterval(() => {
 }, 5000);
 */
 
-const updateIcon = async (tabId, changeInfo, tabInfo) => {
-  if (!changeInfo.url) {
+const extractLightningDataFromPage = async (tabId, changeInfo, tabInfo) => {
+  if (changeInfo.status !== "complete") {
     return;
   }
-  if (!changeInfo.url.startsWith("http")) {
+
+  // delay execution a bit to give the website chance to potentially load async data (e.g. most data from twitter is loaded async)
+  setTimeout(() => {
+    browser.tabs.executeScript(tabId, {
+      code: "if ((document.location.protocol === 'https:' || document.location.protocol === 'http:') && window.LBE_EXTRACT_LIGHTNING_DATA) { LBE_EXTRACT_LIGHTNING_DATA(); };",
+    });
+  }, 500);
```

Comment:
This can easily become buggy with slow connections etc. Maybe better to check with an interval (for e.g. every 100ms) inside a battery if the html contains certain data and if so, cancel the interval.